app/scripts/modules/core/src/forms/uiSelect.decorator.js

```diff
@@ -2,25 +2,23 @@
 
 module.exports = function($provide) {
   $provide.decorator('uiSelectMultipleDirective', function($delegate) {
-    var uiSelect = $delegate[0];
-
-    const originalLink = uiSelect.link;
-
-    uiSelect.compile = function() {
-      return function(scope, element, attrs, ctrls) {
-        originalLink.apply(this, arguments);
-        let $select = ctrls[0];
-        scope.$$listeners['uis:select'] = [];
-        scope.$on('uis:select', function(event, item) {
-          if ($select.selected.length >= $select.limit) {
-            return;
-          }
-          if (!event.defaultPrevented) {
-            $select.selected.push(item);
-            scope.$selectMultiple.updateModel();
-          }
-        });
-      };
+    const [uiSelect] = $delegate,
+      originalLink = uiSelect.link,
+      SELECT_EVENT_KEY = 'uis:select';
+
+    uiSelect.link = function(scope, element, attrs, ctrls) {
```

Comment:
this is a part of angular I don't know much about. What's the distinction between a `.compile` and a `.link` function? It looke like the function is doing the same thing in both cases.